### PR TITLE
RavenDB-11503 fix (including core review related changes, cherry pick from v4.1 branch) 

### PR DESCRIPTION
**Pull request by myarichuk**

### CODE REVIEW

**Comment by ppekrol** [comment]
@myarichuk Can you check failing tests?

File: test/Tests.Infrastructure/RavenTestBase.cs [hunk 1]
```diff
@@ -607,6 +608,7 @@ public class Options
             private Action<DatabaseRecord> _modifyDatabaseRecord;
             private Func<string, string> _modifyDatabaseName;
             private string _path;
+            private bool _isExternalReplicationEnabledForTests;
```
Comment (ppekrol) [inline test/Tests.Infrastructure/RavenTestBase.cs:611]: This is not needed, right?

Comment (myarichuk) [reply test/Tests.Infrastructure/RavenTestBase.cs:611]: Yes. Forgot to remove this I think.

**Comment by myarichuk** [comment]
Finally reproduced the failing tests. Working on it.

**Comment by ppekrol** [comment]
Will be handled here: https://github.com/ravendb/ravendb/pull/7371

**Pull request closed without merging**